input-output-hk/lace
# [LW-9145] TxSummary refactor

# Checklist

- [x] JIRA - LW-9145 LW-9710

---

## Proposed solution
- Refactor the transaction summary for dapps 
- add to and from sections 
- txType and origin 

## Testing
- Run the app 
- Go to wingriders and try to get some wingrider token from ada, there shouldn't be tokens 
- Do the reverse transactions, you should see the token 
- Go to handle.me , mint a token, you should see the transaction 


**Tx notes:** 
- A transaction doesn't have to have to and from addresses. Sometimes it may have one sections with various, and sometimes both. But not always. 
- Transactions from swapping places such as Wingrider or SundaeSwap work with a batcher which means that you might see an amount in their interface, and a slightly different amount of ADA or tADA in the dapp, because you will see the amount of coins that don't get sent to a batcher to process the following transaction to actually create the swap. 


## Screenshots

![Screenshot 2024-02-21 at 08 28 02](https://github.com/input-output-hk/lace/assets/4052063/e965b609-dcb7-4d2e-a92f-4546d70f37dd)
![Screenshot 2024-02-21 at 08 29 17](https://github.com/input-output-hk/lace/assets/4052063/7a3e440e-ef53-447e-8f9d-8885a914ef1d)
![Screenshot 2024-02-21 at 08 33 55](https://github.com/input-output-hk/lace/assets/4052063/d24ea5a3-5bfb-496a-a4ef-52e1c53fa2e3)

// This image is here only to show the display of to address / from address sections, the styles apply from the previous snapshots. 
<img width="400" alt="Screenshot 2024-02-14 at 08 35 05" src="https://github.com/input-output-hk/lace/assets/4052063/81b69b8a-fa43-4349-a89e-b0e069788b07">
